src/BuildScriptGeneratorCli/Commands/DetectCommand.cs

```diff
@@ -0,0 +1,123 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.Detector;
+using Newtonsoft.Json;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Detect all platforms and versions in the given app source directory.")]
+    internal class DetectCommand : CommandBase
+    {
+        public const string Name = "detect";
+
+        [Argument(0, Description = "The source directory. If no value is provided, the current directory is used.")]
+        [DirectoryExists]
+        public string SourceDir { get; set; }
+
+        [Option("--json", Description = "Output the detected platform data in JSON format.")]
+        public bool OutputJson { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<DetectCommand>();
+            var sourceRepo = new LocalSourceRepo(SourceDir, loggerFactory);
+            var ctx = new DetectorContext
+            {
+                SourceRepo = sourceRepo,
+            };
+
+            var detector = serviceProvider.GetRequiredService<IDetector>();
+            var detectedPlatforms = detector.GetAllDetectedPlatforms(ctx);
+            using (var timedEvent = logger.LogTimedEvent("DetectCommand"))
+            {
+                if (detectedPlatforms != null && detectedPlatforms.Any())
+                {
+                    if (OutputJson)
+                    {
+                        console.WriteLine("Detection result in Json format:");
+                        console.WriteLine(JsonFormatResult(detectedPlatforms));
+                    }
+                    else
+                    {
+                        console.WriteLine("Detection result:");
+                        console.WriteLine(ListFormatResult(detectedPlatforms));
+                    }
+
+                    return ProcessConstants.ExitSuccess;
+                }
+            }
+
+            return ProcessConstants.ExitFailure;
+        }
+
+        internal override bool IsValidInput(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<DetectCommand>>();
+            SourceDir = string.IsNullOrEmpty(SourceDir) ? Directory.GetCurrentDirectory() : Path.GetFullPath(SourceDir);
+
+            if (!Directory.Exists(SourceDir))
+            {
+                logger?.LogError("Could not find the source directory.");
+                console.WriteErrorLine($"Could not find the source directory: '{SourceDir}'");
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private string ListFormatResult(IEnumerable<PlatformDetectorResult> detectedPlatforms)
```

Comment:
instead of adding it to string builds, parse it into a string, to then print it, how about call it `PrintTableResult` for example and just write directly to the console?